client/src/services/discordServer.ts

```diff
@@ -1,27 +1,33 @@
-import axios from 'axios';
+import axios, { AxiosInstance } from 'axios';
 import { DiscordServer } from './models';
 
 type DiscordServerResponse = { data: DiscordServer };
 type DiscordServersResponse = { data: DiscordServer[] };
 
 export class DiscordServerService {
-  async deleteDiscordServer(id: number) {
-    const result = await axios.delete<DiscordServerResponse>(`/api/discord-server/${id}`);
-    return result.data.data;
-  }
+  private axios: AxiosInstance;
 
-  async updateDiscordServer(id: number, data: Partial<DiscordServer>) {
-    const result = await axios.put<DiscordServerResponse>(`/api/discord-server/${id}`, data);
-    return result.data.data;
+  constructor() {
+    this.axios = axios.create({ baseURL: `/api/v2` });
   }
 
   async createDiscordServer(data: Partial<DiscordServer>) {
-    const result = await axios.post<DiscordServerResponse>(`/api/discord-server/`, data);
-    return result.data.data;
+    const result = await this.axios.post<DiscordServerResponse>(`/discord-servers`, data);
+    return result.data;
   }
 
   async getDiscordServers() {
-    const result = await axios.get<DiscordServersResponse>(`/api/discord-server`);
-    return result.data.data;
+    const result = await this.axios.get<DiscordServersResponse>(`/discord-servers`);
+    return result.data;
+  }
+
+  async updateDiscordServer(id: number, data: Partial<DiscordServer>) {
+    const result = await this.axios.put<DiscordServerResponse>(`/discord-servers/${id}`, data);
+    return result.data;
+  }
+
+  async deleteDiscordServer(id: number) {
+    const result = await this.axios.delete<DiscordServerResponse>(`/discord-servers/${id}`);
+    return result.data;
```

Comment:
there is a command `npm run openapi` inside nestjs. once you run it, you will be able to use generated methods instead of writing you own. generated files should be commited.
consider checking `notifications.ts` in client for reference